### PR TITLE
calamares: 3.2.17.1 -> 3.2.35.1

### PR DESCRIPTION
**Pull request by freezeboy**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107749` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>calamares</li>
  </ul>
</details>

cmake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

`patchPhase` should not be overridden, use `postPatch` instead.

**Comment by freezeboy** [comment]
Thanks for the review

File: pkgs/tools/misc/calamares/default.nix [hunk 1]
```diff
@@ -6,16 +6,17 @@
...
   src = fetchurl {
     url = "https://github.com/${pname}/${pname}/releases/download/v${version}/${pname}-${version}.tar.gz";
-    sha256 = "156zpjyw8w4y23aa60mvg3d3mr0kzfq5jkl7ixgahq33zpc17ms8";
+    sha256 = "s2wnwcdrcJLG5NhugSkntBCYfPuv3T/9+PclbmK0BJ4=";
   };
 
+  nativeBuildInputs = [ cmake ];
   buildInputs = [
-    boost cmake extra-cmake-modules kparts.dev kpmcore.out kservice.dev
+    boost extra-cmake-modules kparts.dev kpmcore.out kservice.dev
```
Comment (marsam) [inline pkgs/tools/misc/calamares/default.nix:19]: would you mind also moving `extra-cmake-modules` to `nativeBuildInputs`?

Comment (freezeboy) [reply pkgs/tools/misc/calamares/default.nix]: No problem, done

File: pkgs/tools/misc/calamares/default.nix [hunk 2]
```diff
@@ -6,16 +6,17 @@
...
+  version = "3.2.35.1";
 
   # release including submodule
   src = fetchurl {
     url = "https://github.com/${pname}/${pname}/releases/download/v${version}/${pname}-${version}.tar.gz";
-    sha256 = "156zpjyw8w4y23aa60mvg3d3mr0kzfq5jkl7ixgahq33zpc17ms8";
+    sha256 = "s2wnwcdrcJLG5NhugSkntBCYfPuv3T/9+PclbmK0BJ4=";
   };
 
+  nativeBuildInputs = [ cmake cmake-extra-modules ];
```
Comment (marsam) [inline pkgs/tools/misc/calamares/default.nix:17]:
```suggestion
  nativeBuildInputs = [ cmake extra-cmake-modules ];
```

Comment (freezeboy) [reply pkgs/tools/misc/calamares/default.nix]: oops

**Pull request merged**